ScheduleLesson/Controllers/ScheduleController.cs

```diff
@@ -33,19 +38,24 @@ public async Task<ActionResult<Schedule>> GetSchedule(int id)
         [HttpPost]
         public async Task<ActionResult<Schedule>> AddSchedule(Schedule schedule)
         {
-            Schedule result = await _scheduleService.AddSchedule(schedule);
+            string? userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            Schedule result = await _scheduleService.AddSchedule(schedule, userId);
             return Ok(result);
         }
+
         [HttpPut]
         public async Task<ActionResult<Schedule>> UpdateSchedule(Schedule Id)
         {
-            Schedule? dbScheduleUpdated = await _scheduleService.UpdateSchedule(Id);
+            string? userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            Schedule? dbScheduleUpdated = await _scheduleService.UpdateSchedule(Id, userId);
             return Ok("Updated Successfully");
         }
+
         [HttpDelete("{id}")]
         public async Task<ActionResult<List<Schedule>>> DeleteSchedule(int id)
         {
-            Schedule? scheduleDeleted = await _scheduleService.DeleteSchedule(id);
```

Comment:
It seems that only the id can be used here, since we can identify a unique schedule by its id.

Reply:
You are right: the id is unique for each Schedule, and there is no need to pass the userGuid to the DeleteSchedule method. However, if I want to ensure that a user can only delete their own schedules, then using the userGuid is justified. It adds a layer of security so that one user cannot delete another's schedule.